beets/util/artresizer.py

```diff
@@ -173,6 +173,46 @@ def __init__(self):
         log.debug(u"artresizer: method is {0}", self.method)
         self.can_compare = self._can_compare()
 
+    def valid_size(self, size, enforce_ratio = False, minwidth = None):
+        """If size constraints exist, check whether the provided image size
+        matches them.
+        """
+        # Check minimum size.
+        if minwidth and size[0] < minwidth:
+            log.debug('image too small ({} < {})',
+                      size[0], minwidth)
+            return False
+
+        # Check aspect ratio.
+        if enforce_ratio and size[0] != size[1]:
+            log.debug('image is not square ({} != {})',
+                       size[0], size[1])
+            return False
+
+        return True
+
+    def must_resize(self, size, maxwidth = None):
+        """Determine whether the provided image size means that the image
+        will need to be scaled to fit the maximum width.
+        """
+        if not maxwidth:
+            return False
+
+        if not size:
+            log.warning(u'Could not get size of image (please see '
+                        u'documentation for dependencies).')
```

Comment:
It wasn't true for the new case, but as you say is incomplete compared to the old -- maybe it can return None and I can move that logging to the plugins.